File: coyote/models/get_namespace_out.py
# this file is @generated
from pydantic import Field
from datetime import datetime

from ..internal.base_model import BaseModel

from .retention import Retention
from .storage_type import StorageType


class GetNamespaceOut(BaseModel):
    created: datetime

    name: str

    retention: Retention

    storage_type: StorageType = Field(alias="storage_type")

    updated: datetime

File: src/models/get_namespace_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{retention::Retention, storage_type::StorageType};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct GetNamespaceOut {
    pub created: jiff::Timestamp,

    pub name: String,

    pub retention: Retention,

    pub storage_type: StorageType,

    pub updated: jiff::Timestamp,
}

impl GetNamespaceOut {
    pub fn new(
        created: jiff::Timestamp,
        name: String,
        retention: Retention,
        storage_type: StorageType,
        updated: jiff::Timestamp,
    ) -> Self {
        Self {
            created,
            name,
            retention,
            storage_type,
            updated,
        }
    }
}
